hswidget/fonttablewidget/widget.go

```diff
@@ -0,0 +1,352 @@
+package fonttablewidget
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/ianling/giu"
+
+	"github.com/OpenDiablo2/OpenDiablo2/d2common/d2fileformats/d2font"
+	"github.com/OpenDiablo2/OpenDiablo2/d2common/d2fileformats/d2font/d2fontglyph"
+)
+
+const (
+	inputIntW                = 30
+	delW, delH               = 50, 30
+	upW, upH                 = 30, 30
+	downW, downH             = 40, 30
+	addW, addH               = 400, 30
+	editRuneW, editRuneH     = 50, 30
+	saveCancelW, saveCancelH = 80, 30
+)
+
+type textures struct {
+	up, down, del *giu.Texture
+}
+
+// widget represents a font table widget
+type widget struct {
+	fontTable *d2font.Font
+	id        string
+	textures
+}
+
+// Create creates a new FontTable widget
+func Create(
+	up, down, del *giu.Texture,
+	id string, fontTable *d2font.Font,
+) giu.Widget {
```

Comment:
I think in the future we should try to leave these textures out of the provider function for the widget. These are more or less hardcoded, so why not just keep these inside of the widget impl?

If you look at the widgets exported by `giu`, I think you'll see a lot of them will take a uid and some other widget parameters in the provider func; these textures are not really parameters in the sense that they change widget operation, they are literally being passed in to be assigned somewhere.

tldr; we should manage textures inside of the widget, not pass them into the provider func.

Reply:
it doesn't work;
when we place texture loading in widget, image buttons aren't displayed.

Reply:
maybe, we can place it in widget's state. What do you think about it?

Reply:
Yeah, we might be able to, but it seems almost like an app-level piece of functionality (loading single texture to be used everywhere). Perhaps we can come up with an enum for the texture ID's + a mapping of paths to ID's. Then we might be able to put all the texture loading stuff somewhere else and not worry about that in the widget code.

anyways, definitely another PR.